rllib/policy/tests/test_policy_state_swapping.py

```diff
@@ -0,0 +1,129 @@
+import gym
+import numpy as np
+import tree  # pip install dm_tree
+import unittest
+
+import ray
+from ray.rllib.algorithms.appo import (
+    APPOConfig,
+    APPOTF1Policy,
+    APPOTF2Policy,
+    APPOTorchPolicy,
+)
+from ray.rllib.policy.policy_map import PolicyMap
+from ray.rllib.utils.framework import try_import_tf
+from ray.rllib.utils.test_utils import check, framework_iterator
+from ray.rllib.utils.tf_utils import get_tf_eager_cls_if_necessary
+
+tf1, tf, tfv = try_import_tf()
+
+
+class TestPolicyStateSwapping(unittest.TestCase):
+    """Tests, whether Policies can be "swapped out" via their state on a GPU."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        ray.init()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        ray.shutdown()
+
+    def test_policy_swap_gpu(self):
+        config = APPOConfig().framework("tf2", eager_tracing=True).resources(num_gpus=1)
+        obs_space = gym.spaces.Box(-1.0, 1.0, (4,), dtype=np.float32)
+        dummy_obs = obs_space.sample()
+        act_space = gym.spaces.Discrete(2)
+        num_policies = 2
+        capacity = 1
+
+        for fw in framework_iterator(config):
+            cls = get_tf_eager_cls_if_necessary(
+                APPOTF2Policy
+                if fw == "tf2"
+                else APPOTF1Policy
+                if fw == "tf"
+                else APPOTorchPolicy,
+                config,
+            )
+
+            # Create empty, swappable-policies PolicyMap.
+            policy_map = PolicyMap(capacity=capacity, policies_swappable=True)
+
+            # Create and add some TF2 policies.
+            for i in range(num_policies):
+                config.training(lr=(i + 1) * 0.00001)
+                with tf1.variable_scope(f"Policy{i}"):
+                    policy = cls(
+                        observation_space=obs_space,
+                        action_space=act_space,
+                        config=config.to_dict(),
+                    )
+                policy_map[f"pol{i}"] = policy
+
+            dummy_batch = tree.map_structure(
```

Comment:
Because this one here is a loss/learning update test. So I need an entire sample batch incl actions, rewards, dones, etc..
In the PolicyMap test, afair, I'm just testing action computations with observations as input.